### ISSUE TITLE
训练问题

### ISSUE THREAD

**Issue by timelovery** [tianrun-chen/SAM-Adapter-PyTorch#126]
## train log
poch 169/200, train G: loss=1.0255, val: sm=0.6636, val: em=0.5854, val: wfm=0.6213, val: mae=0.1898, 1.7m 4.8h/5.7h                                                                                                    
epoch 170/200, train G: loss=1.0265, val: sm=0.6728, val: em=0.5950, val: wfm=0.6210, val: mae=0.1883, 1.7m 4.8h/5.7h                                                                                                    
epoch 171/200, train G: loss=1.0286, val: sm=0.6739, val: em=0.5958, val: wfm=0.6212, val: mae=0.1881, 1.7m 4.8h/5.7h                                                                                                    
epoch 172/200, train G: loss=1.0231, val: sm=0.6646, val: em=0.5862, val: wfm=0.6219, val: mae=0.1895, 1.7m 4.9h/5.7h                                                                                                    
epoch 173/200, train G: loss=1.0275, val: sm=0.6659, val: em=0.5880, val: wfm=0.6217, val: mae=0.1893, 1.7m 4.9h/5.7h                                                                                                    
epoch 174/200, train G: loss=1.0238, val: sm=0.6744, val: em=0.5956, val: wfm=0.6211, val: mae=0.1880, 1.7m 4.9h/5.7h                                                                                                    
epoch 175/200, train G: loss=1.0236, val: sm=0.6710, val: em=0.5918, val: wfm=0.6198, val: mae=0.1889, 1.7m 5.0h/5.7h                                                                                                    
epoch 176/200, train G: loss=1.0238, val: sm=0.6677, val: em=0.5893, val: wfm=0.6222, val: mae=0.1890, 1.7m 5.0h/5.7h                                                                                                    
epoch 177/200, train G: loss=1.0261, val: sm=0.6823, val: em=0.6061, val: wfm=0.6233, val: mae=0.1861, 1.7m 5.0h/5.7h                                                                                                    
epoch 178/200, train G: loss=1.0251, val: sm=0.6721, val: em=0.5940, val: wfm=0.6215, val: mae=0.1883, 1.7m 5.0h/5.7h                                                                                                    
epoch 179/200, train G: loss=1.0253, val: sm=0.6553, val: em=0.5757, val: wfm=0.6168, val: mae=0.1920, 1.7m 5.1h/5.7h                                                                                                    
epoch 180/200, train G: loss=1.0226, val: sm=0.6783, val: em=0.5994, val: wfm=0.6205, val: mae=0.1875, 1.7m 5.1h/5.7h                                                                                                    
epoch 181/200, train G: loss=1.0230, val: sm=0.6615, val: em=0.5820, val: wfm=0.6180, val: mae=0.1908, 1.7m 5.1h/5.7h                                                                                                    
epoch 182/200, train G: loss=1.0238, val: sm=0.6684, val: em=0.5902, val: wfm=0.6215, val: mae=0.1890, 1.7m 5.2h/5.7h                                                                                                    
epoch 183/200, train G: loss=1.0246, val: sm=0.6664, val: em=0.5873, val: wfm=0.6184, val: mae=0.1899, 1.7m 5.2h/5.7h                                                                                                    
epoch 184/200, train G: loss=1.0233, val: sm=0.6723, val: em=0.5926, val: wfm=0.6189, val: mae=0.1888, 1.7m 5.2h/5.7h                                                                                                    
epoch 185/200, train G: loss=1.0241, val: sm=0.6676, val: em=0.5874, val: wfm=0.6169, val: mae=0.1900, 1.7m 5.2h/5.7h                                                                                                    
epoch 186/200, train G: loss=1.0237, val: sm=0.6597, val: em=0.5810, val: wfm=0.6193, val: mae=0.1908, 1.7m 5.3h/5.7h                                                                                                    
epoch 187/200, train G: loss=1.0224, val: sm=0.6556, val: em=0.5762, val: wfm=0.6175, val: mae=0.1919, 1.7m 5.3h/5.7h                                                                                                    
epoch 188/200, train G: loss=1.0260, val: sm=0.6720, val: em=0.5944, val: wfm=0.6222, val: mae=0.1882, 1.7m 5.3h/5.7h                                                                                                    
epoch 189/200, train G: loss=1.0228, val: sm=0.6746, val: em=0.5966, val: wfm=0.6221, val: mae=0.1878, 1.7m 5.3h/5.7h                                                                                                    
epoch 190/200, train G: loss=1.0263, val: sm=0.6698, val: em=0.5916, val: wfm=0.6211, val: mae=0.1888, 1.7m 5.4h/5.7h                                                                                                    
epoch 191/200, train G: loss=1.0231, val: sm=0.6748, val: em=0.5958, val: wfm=0.6193, val: mae=0.1883, 1.7m 5.4h/5.7h                                                                                                    
epoch 192/200, train G: loss=1.0240, val: sm=0.6672, val: em=0.5873, val: wfm=0.6176, val: mae=0.1899, 1.7m 5.4h/5.7h                                                                                                    
epoch 193/200, train G: loss=1.0229, val: sm=0.6724, val: em=0.5931, val: wfm=0.6187, val: mae=0.1888, 1.7m 5.5h/5.7h                                                                                                    
epoch 194/200, train G: loss=1.0225, val: sm=0.6729, val: em=0.5938, val: wfm=0.6193, val: mae=0.1886, 1.7m 5.5h/5.7h                                                                                                    
epoch 195/200, train G: loss=1.0223, val: sm=0.6673, val: em=0.5876, val: wfm=0.6177, val: mae=0.1899, 1.7m 5.5h/5.7h                                                                                                    
epoch 196/200, train G: loss=1.0216, val: sm=0.6647, val: em=0.5848, val: wfm=0.6174, val: mae=0.1903, 1.7m 5.5h/5.7h                                                                                                    
epoch 197/200, train G: loss=1.0256, val: sm=0.6724, val: em=0.5942, val: wfm=0.6209, val: mae=0.1884, 1.7m 5.6h/5.7h                                                                                                    
epoch 198/200, train G: loss=1.0230, val: sm=0.6629, val: em=0.5846, val: wfm=0.6201, val: mae=0.1901, 1.7m 5.6h/5.7h                                                                                                    
epoch 199/200, train G: loss=1.0232, val: sm=0.6637, val: em=0.5850, val: wfm=0.6193, val: mae=0.1902, 1.7m 5.6h/5.7h                                                                                                    
epoch 200/200, train G: loss=1.0230, val: sm=0.6676, val: em=0.5891, val: wfm=0.6199, val: mae=0.1894, 1.7m 5.7h/5.7h   

## infer

<img width="528" height="469" alt="Image" src="https://github.com/user-attachments/assets/610336ea-1d70-4dc0-bbf7-4ea74b28754f" />

## trian yaml
train_dataset:
  dataset:
    name: paired-image-folders
    args:
      root_path_1: load/CAMO1/Images/Train
      root_path_2: load/CAMO1/GT
      cache: none
      split_key: train
  wrapper:
    name: train
    args:
      inp_size: 1024
      augment: true
  batch_size: 1

val_dataset:
  dataset:
    name: paired-image-folders
    args:
      root_path_1: load/CAMO1/Images/Test
      root_path_2: load/CAMO1/GT
      cache: none
      split_key: test
  wrapper:
    name: val
    args:
      inp_size: 1024
  batch_size: 1

test_dataset:
  dataset:
    name: paired-image-folders
    args:
      root_path_1: load/CAMO1/Images/Test
      root_path_2: load/CAMO1/GT
      cache: none
      split_key: test
  wrapper:
    name: val
    args:
      inp_size: 1024
  batch_size: 1

eval_type: cod
sam_checkpoint: ./pretrained/sam_vit_b_01ec64.pth
data_norm:
  inp:
    sub:
    - 0.5
    div:
    - 0.5
  gt:
    sub:
    - 0.5
    div:
    - 0.5
  gt_rgb:
    sub:
    - 0.5
    div:
    - 0.5
model:
  name: sam
  args:
    inp_size: 1024
    loss: iou
    encoder_mode:
      name: sam
      img_size: 1024
      mlp_ratio: 4
      patch_size: 16
      qkv_bias: true
      use_rel_pos: true
      window_size: 14
      out_chans: 256
      scale_factor: 32
      input_type: fft
      freq_nums: 0.25
      prompt_type: highpass
      prompt_embed_dim: 256
      tuning_stage: 1234
      handcrafted_tune: true
      embedding_tune: true
      adaptor: adaptor
      embed_dim: 768
      depth: 12
      num_heads: 12
      global_attn_indexes:
      - 2
      - 5
      - 8
      - 11
optimizer:
  name: adamw
  args:
    lr: 0.00001
lr_min: 1.0e-7
epoch_max: 200

multi_step_lr:
  milestones:
  - 1
  gamma: 0.1
epoch_val: 1
epoch_save: 1

#resume: 60
#start_epoch: 60

## Train data

<img width="527" height="638" alt="Image" src="https://github.com/user-attachments/assets/f10e8f1e-64bd-4813-9d6a-3f575da61129" />

## 
请问是我训练数据存放不对，还是有什么处理没有做，为什么训练完全没有学到东西呢

**Comment by hellokhj** [comment]
我也是这样子 请问解决了吗
